bftengine/include/bftengine/Replica.hpp

```diff
@@ -50,6 +55,8 @@ class IRequestsHandler {
 
   virtual void onFinishExecutingReadWriteRequests() {}
   virtual ~IRequestsHandler() {}
+
+  virtual std::shared_ptr<IControlHandler> getControlHandlers() = 0;
```

Comment:
I'm a bit confused about why this is part of the interface. Doing this requires all users to pass in an `IControlHandler`, as evidenced by the required modification to tests in the code below. However, all these tests simply return a nullptr, which means that all users of this interface now need to check for a nullptr. It seems to me that every real system should have a set of control handlers, and that the reference should never be null. Also, I don't see where the request handler gets passed in. If it's at a higher level of abstraction, then why not also put the setter there. As with another comment further down, I think we may want to revisit our abstractions.

Reply:
Like above, this makes sense in light of the fact that the command handlers are set at a higher level of abstraction, but used inside concord-bft. This is how they are accessed from concord-bft.